lib/iris/analysis/cartography.py

```diff
@@ -599,6 +599,12 @@ def project(cube, target_proj, nx=None, ny=None):
         on the statistics of the data e.g. the mean and standard deviation
         will not be preserved.
 
+    .. warning::
+
+        If the target projection is non-rectangular, e.g. Robinson, the target
+        grid may include points outside the boundary of the projection. The
+        latitude/longitude of such points may be unpredictable.
+
```

Comment:
Such a warning should have already been present, but has become more apparent with the work done on this PR with the Robinson projection 👍